infra/aws/lambdas/ECSServiceRedployFucntion.py

```diff
@@ -0,0 +1,48 @@
+#  Copyright (c) 2024 Talent Beyond Boundaries.
+#
+#  This program is free software: you can redistribute it and/or modify it under
+#  the terms of the GNU Affero General Public License as published by the Free
+#  Software Foundation, either version 3 of the License, or any later version.
+#
+#  This program is distributed in the hope that it will be useful, but WITHOUT
+#  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+#  FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+#  for more details.
+#
+#  You should have received a copy of the GNU Affero General Public License
+#  along with this program. If not, see https://www.gnu.org/licenses/.
+
+import boto3
+import json
+from datetime import datetime
+
+
+def lambda_handler(event, context):
+    # Initialize ECS client using Boto3 to interact with ECS service
+    ecs_client = boto3.client('ecs')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

A Boto3 client is initialized from a Lambda handler. To speed up your Boto3 client initialization and minimize the operational cost of your Lambda function, we recommend that you create the client at the level of the module that contains the handler, and then reuse it between invocations.

[Learn more](https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html#function-code)

Suggested remediation:
We recommended the following changes to mitigate the issue.
```
@@ -20,5 +21,5 @@
 def lambda_handler(event, context):
     # Initialize ECS client using Boto3 to interact with ECS service
-    ecs_client = boto3.client(&#39;ecs&#39;)
+    
 
     # Define the ECS cluster and service to be updated

```

Reply:
We only call the boto3 client once so pre initialising doesn't save us any operational cost in the case of this lambda.